crates/asdf-plugin/src/lib.rs

```diff
@@ -0,0 +1,221 @@
+    }
+
+    pub fn download_prebuilt(&self, input: DownloadPrebuiltInput) -> DownloadPrebuiltOutput {
+        let env = get_host_environment().unwrap();
+
+        check_supported_os_and_arch(
+            "ASDF Plugin",
+            &env,
+            permutations![
+                HostOS::Linux => [HostArch::X64, HostArch::Arm64, HostArch::Arm, HostArch::Powerpc64, HostArch::S390x],
+                HostOS::MacOS => [HostArch::X64, HostArch::Arm64],
+                HostOS::Windows => [HostArch::X64, HostArch::X86, HostArch::Arm64],
+            ],
+        ).unwrap();
+
+        let version = input.context.version;
+        let arch = env.arch;
+        let os = env.os;
+
+        let prefix = match os {
+            HostOS::Linux => format!("asdf-plugin-v{version}-linux-{arch}"),
+            HostOS::MacOS => format!("asdf-plugin-v{version}-darwin-{arch}"),
+            HostOS::Windows => format!("asdf-plugin-v{version}-win-{arch}"),
+            other => {
+                return DownloadPrebuiltOutput {
+                    download_url: format!("Unsupported platform: {}", other),
+                    ..DownloadPrebuiltOutput::default()
+                };
+            }
+        };
+
+        let filename = if os == HostOS::Windows {
+            format!("{prefix}.zip")
+        } else {
+            format!("{prefix}.tar.xz")
+        };
+
+        let config: AsdfConfig = self.tool.config();
+        let asdf_plugin = config
+            .asdf_plugin
+            .unwrap_or_else(|| self.tool.get_name().to_string());
+        let repository = config
+            .asdf_repository
+            .unwrap_or_else(|| format!("https://github.com/asdf-vm/asdf-{}.git", asdf_plugin));
+
+        DownloadPrebuiltOutput {
+            archive_prefix: Some(prefix),
+            download_url: format!("{repository}/releases/download/v{version}/{filename}"),
+            download_name: Some(filename),
+            checksum_url: Some(format!(
+                "{repository}/releases/download/v{version}/SHA256SUMS"
+            )),
+            checksum_public_key: Some("public-key-string".into()), // Need to adjust if applicable
+            ..DownloadPrebuiltOutput::default()
+        }
+    }
+
+    pub fn install_plugin(&self, repository: &str) {
+        let asdf_dir = match env::var("ASDF_DATA_DIR") {
+            Ok(val) => Path::new(&val).to_path_buf(),
+            Err(_) => dirs::home_dir().unwrap().join(".asdf"),
+        };
+
+        let plugin_dir = asdf_dir.join("plugins").join(self.tool.get_name());
+
+        if !plugin_dir.exists() {
+            std::fs::create_dir_all(&plugin_dir).unwrap();
+        }
+
+        std::process::Command::new("git")
+            .arg("clone")
+            .arg(repository)
+            .arg(&plugin_dir)
+            .output()
+            .expect("Failed to clone asdf plugin");
+    }
+
+    pub fn pre_install(&self, mut input: InstallHook) {
+        let config: AsdfConfig = self.tool.config();
```

Comment:
I had confusion here about referencing config. Any guidance on this?

Reply:
Use https://docs.rs/proto_pdk/latest/proto_pdk/fn.get_tool_config.html